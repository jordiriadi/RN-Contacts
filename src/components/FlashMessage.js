import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Text, View, StyleSheet } from 'react-native';
import Styles from '../utils/Styles';

const FlashMessage = (props) => {
  const { text, onTimeout } = props;
  const [open, setOpen] = useState(false);

  useEffect(() => {
    setOpen(true);
  }, [])

  useEffect(() => {
    const timer = setTimeout(() => closeFlashMessage(), 3000);

    return () => {
      clearTimeout(timer);
    }
  }, [open])

  const closeFlashMessage = () => {
    setOpen(false);
    onTimeout();
  }

  return (
    <View style={styles.container}>
      <Text style={{ color: 'white' }}>{text}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    backgroundColor: 'orangered',
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    height: Styles.DefaultMargin * 5,
    padding: Styles.DefaultPadding,
    top: 0
  }
});

FlashMessage.propTypes = {
  text: PropTypes.string,
  onTimeout: PropTypes.func,
}

export default FlashMessage;