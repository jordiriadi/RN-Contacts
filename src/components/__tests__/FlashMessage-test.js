import 'react-native';
import React from 'react';
import {shallow} from 'enzyme';

import FlashMessage from '../FlashMessage';

describe('FlashMessage Component', () => {  
  it('renders correctly', () => {
  
    const wrapper = shallow(<FlashMessage/>);

    expect(wrapper).toMatchSnapshot();
  });

})