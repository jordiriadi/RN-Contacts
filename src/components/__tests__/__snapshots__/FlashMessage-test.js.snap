// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlashMessage Component renders correctly 1`] = `ShallowWrapper {}`;
